core/src/main/c/share/jit/avx2.h

```diff
@@ -205,10 +208,45 @@ namespace questdb::avx2 {
         return {length_data, data_type_t::i64, data_kind_t::kMemory};
     }
 
+    // Reads length part of the varchar header for aux vector.
+    // This part is stored in the lowest 4 bytes of the header
+    // (see VarcharTypeDriver to understand the format).
+    //
+    // Note: unlike read_mem_varlen this method doesn't return the length,
+    //       so it can only be used in NULL checks.
+    jit_value_t read_mem_varchar_len_header(Compiler &c,
+                                            int32_t column_idx,
+                                            const Gp &varsize_indexes_ptr,
+                                            const Gp &input_index) {
+        Gp varsize_index_address = c.newInt64("varsize_index_address");
+        c.mov(varsize_index_address, ptr(varsize_indexes_ptr, 8 * column_idx, 8));
+
+        auto header_shift = type_shift(data_type_t::i128);
+        auto header_size = 1 << header_shift;
+        Gp header_offset = c.newInt64("header_offset");
+        c.mov(header_offset, input_index);
+        c.sal(header_offset, header_shift);
+
+        Ymm header_data = c.newYmm();
+        c.vmovdqu(header_data, ymmword_ptr(varsize_index_address, header_offset, 0));
+
+        // Mask higher bits since they store the data vector offset
+        // and we don't need it in null checks.
+        Ymm header_masks = c.newYmm("header_masks");
+        // Erase 64 highest bits (LE) in each aux header value.
+        int64_t masks[4] = {-1, 0, -1, 0};
+        Mem masksMem = c.newConst(ConstPool::kScopeLocal, &masks, 32);
+        c.vmovdqu(header_masks, masksMem);
+        c.vpand(header_data, header_data, header_masks);
+
+        return {header_data, data_type_t::i128, data_kind_t::kMemory};
```

Comment:
Wouldn't it be better to represent the VARCHAR header as 32-bit or 64-bit value, and gather them into a single YMM? It would have less branching instructions per value and it would probably be more useful for combining with other value types (as almost nothing else is 128-bit).